src/metabase/actions/execution.clj

```diff
@@ -75,53 +78,70 @@
         extra-parameters (set/difference (set (keys request-parameters))
                                          (set (keys slug->field-name)))
-
         pk-field (first pk-fields)
         simple-parameters (update-keys request-parameters (comp keyword slug->field-name))
         pk-field-name (keyword (:name pk-field))
-        _ (api/check (empty? extra-parameters)
-                     400
-                     {:message (tru "No destination parameter found for {0}. Found: {1}"
-                                    (pr-str extra-parameters)
-                                    (pr-str (set (keys slug->field-name))))
-                      :parameters request-parameters
-                      :destination-parameters (keys slug->field-name)})
-
-        _ (api/check (or (not requires_pk)
-                         (some? (get simple-parameters pk-field-name)))
-                     400
-                     (tru "Missing primary key parameter: {0}"
-                          (pr-str (u/slugify (:name pk-field)))))
-        query (cond-> {:database database-id,
-                       :type :query,
-                       :query {:source-table table-id}}
-                requires_pk
-                (assoc-in [:query :filter]
-                          [:= [:field (:id pk-field) nil] (get simple-parameters pk-field-name)]))
-        implicit-action (cond
-                          (= slug "delete")
-                          :row/delete
-
-                          requires_pk
-                          :row/update
-
-                          :else
-                          :row/create)
         row-parameters (cond-> simple-parameters
-                         (not= implicit-action :row/create) (dissoc pk-field-name))
+                         (not= implicit-action :row/create) (dissoc pk-field-name))]
+    (api/check (or (not requires_pk)
+                   (some? (get simple-parameters pk-field-name)))
+               400
+               (tru "Missing primary key parameter: {0}"
+                    (pr-str (u/slugify (:name pk-field)))))
+    (api/check (empty? extra-parameters)
+               400
+               {:message (tru "No destination parameter found for {0}. Found: {1}"
+                              (pr-str extra-parameters)
+                              (pr-str (set (keys slug->field-name))))
+                :parameters request-parameters
+                :destination-parameters (keys slug->field-name)})
+    (cond->
+      {:query {:database database-id,
+               :type :query,
+               :query {:source-table table-id}}
+       :row-parameters row-parameters}
+
+      requires_pk
+      (assoc-in [:query :query :filter]
+                [:= [:field (:id pk-field) nil] (get simple-parameters pk-field-name)])
+
+      requires_pk
+      (assoc :prefetch-parameters [{:target [:dimension [:field (:id pk-field) nil]]
+                                    :type "id"
```

Comment:
I'm not sure if type "id" is always sufficient (for varchar pks). Is there a way to go from `:base_type` or `:semantic_type` (`:type/PK :type/BigInteger`) to a `parameter.type` `:number`.